accounts/accounts_cli.go

```diff
@@ -0,0 +1,328 @@
+	if err != nil {
+		return errors.New(fmt.Sprint("Login request failed", err.Error()))
+	}
+
+	switch res.StatusCode {
+	case 200:
+		data, _ := ioutil.ReadAll(res.Body)
+		json.Unmarshal(data, &a.User)
+		a.User.Token = res.Header.Get("x-tidepool-session-token")
+		return nil
+	default:
+		return errors.New(fmt.Sprint("Login failed", res.StatusCode))
+	}
+}
+
+func (a *admin) findAccounts() error {
+
+	urlPath := a.host + fmt.Sprintf("/access/groups/%s", a.User.Id)
+
+	req, _ := http.NewRequest("GET", urlPath, nil)
+	req.Header.Add("x-tidepool-session-token", a.User.Token)
+
+	res, err := a.client.Do(req)
+
+	if err != nil {
+		return errors.New("Could attempt to find administered accounts: " + err.Error())
+	}
+
+	switch res.StatusCode {
+	case 200:
+		data, _ := ioutil.ReadAll(res.Body)
+
+		var raw map[string]interface{}
+
+		json.Unmarshal(data, &raw)
+
+		for key, value := range raw {
+			a.Accounts = append(a.Accounts, &Account{Id: string(key), Perms: value})
+		}
+		return nil
+	default:
+		log.Println("Failed finding profiles", res.StatusCode)
+		return nil
+	}
+}
+
+func (a *admin) loadExistingReport(path string) error {
+	log.Println("loading audit report ...")
+
+	rf, err := os.Open(path)
+	if err != nil {
+		return err
+	}
+
+	jsonParser := json.NewDecoder(rf)
+	if err = jsonParser.Decode(&a); err != nil {
+		return err
+	}
+	rf.Close()
+	return nil
+}
+
+func (a *admin) accountDetails() {
+	var wg sync.WaitGroup
+
+	for _, account := range a.Accounts {
+		// Increment the WaitGroup counter.
+		wg.Add(1)
+		// Launch a goroutine to account data
+		go func(account *Account) {
+			// Decrement the counter when the goroutine completes.
+			defer wg.Done()
+			// Fetch the account data
+			a.findPublicInfo(account)
+		}(account)
+	}
+	// Wait for all fetches to complete.
+	wg.Wait()
+	return
+}
+func (a *admin) audit() {
+
+	var wg sync.WaitGroup
+
+	for _, account := range a.Accounts {
+		// Increment the WaitGroup counter.
+		wg.Add(1)
+		// Launch a goroutine to account data
+		go func(account *Account) {
+			// Decrement the counter when the goroutine completes.
+			defer wg.Done()
+			a.findLastUploaded(account)
+		}(account)
+	}
+	// Wait for all fetches to complete.
+	wg.Wait()
+	return
+}
+
+func setHost(targetEnv string) string {
+
+	targetEnv = strings.ToLower(targetEnv)
+
+	fmt.Println("Run audit against:", targetEnv)
+
+	if targetEnv == "devel" {
+		return develHost
+	} else if targetEnv == "prod" {
+		return prodHost
+	} else if targetEnv == "staging" {
+		return stagingHost
+	}
+	return localHost
+}
+
+// and audit will find all account linked
+func auditAccounts(c *cli.Context) {
+
+	if c.String("username") == "" {
+		log.Fatal("Please specify the username with the --username or -u flag.")
+	}
+
+	adminUser := &admin{host: setHost(c.String("env")), client: &http.Client{}}
+
+	fmt.Printf("Password: ")
+	pass := gopass.GetPasswdMasked()
+
+	err := adminUser.login(c.String("username"), string(pass[:]))
```

Comment:
ahh - god I have been doing it that way for I while and I can't even remember why now :) good to know I will go for the simple option and try and remember why I started doing that